app/controllers/works_controller.rb

```diff
@@ -0,0 +1,75 @@
+class WorksController < ApplicationController
+
+  before_action :find_work, only: [:show, :edit, :update, :destroy]
+  def index
+    @works = Work.order("votes_count DESC, created_at")
+  end
+
+  def show
+    if @work.nil?
+      render file: "#{Rails.root}/public/404.html",  layout: false, status: :not_found
```

Comment:
This totally works, but I'd say a flash message or a custom html page would feel less clunky for a case like this than the generic Rails page that says on it "If you are the application owner check the logs for more information."